okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -1193,6 +1193,40 @@ private static DeploymentDescriptor pickInstance(List<DeploymentDescriptor> inst
         });
   }
 
+  private Future<OkapiClient> doCallSystemInterface2(
+      MultiMap headersIn, String tenantId, String authToken,
+      ModuleInstance inst, String modPerms, String request) {
+
+    Map<String, String> headers = sysReqHeaders(headersIn, tenantId, authToken, inst, modPerms);
+    headers.put(XOkapiHeaders.URL_TO, inst.getUrl());
+    logger.debug("syscall begin {} {}{}", inst.getMethod(), inst.getUrl(), inst.getPath());
+    OkapiClient cli = new OkapiClient(this.httpClient, inst.getUrl(), vertx, headers);
+    String reqId = inst.getPath().replaceFirst("^[/_]*([^/]+).*", "$1");
```

Comment:
```suggestion
    // first path segment, but skip leading /_
    String reqId = inst.getPath().replaceFirst("^(/_)?/([^/]+).*", "$2");
```

Reply:
This replaceFirst was not part of PR (only due to refactorings)..

The initial replaceFirst would make `tenant` from path  `/_/tenant`. Your suggestion ends up with an empty string.

Reply:
`System.out.println("/_/tenant".replaceFirst("^(/_)?/([^/]+).*", "$2"));`
outputs
`tenant`

For all unit tests my suggestion returns the same result as before.